x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapper.java

```diff
@@ -0,0 +1,323 @@
+                    iterator.remove();
+                } else if (propName.equals(Names.NULL_VALUE)) {
+                    if (propNode == null) {
+                        throw new MapperParsingException("Property [null_value] cannot be null.");
+                    }
+                    nullValue = propNode;
+                    iterator.remove();
+                }
+            }
+
+            if (nullValue != null) {
+                boolean ignoreMalformed = builder.ignoreMalformed == null ? Defaults.IGNORE_MALFORMED.value() : builder.ignoreMalformed;
+                CartesianPoint point = CartesianPoint.parsePoint(nullValue);
+                if (ignoreMalformed == false) {
+                    if (Float.isFinite(point.getX()) == false) {
+                        throw new IllegalArgumentException("illegal x value [" + point.getX() + "]");
+                    }
+                    if (Float.isFinite(point.getY()) == false) {
+                        throw new IllegalArgumentException("illegal y value [" + point.getY() + "]");
+                    }
+                }
+                builder.nullValue(point);
+            }
+            return builder;
+        }
+    }
+
+    protected Explicit<Boolean> ignoreMalformed;
+
+    public PointFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType,
+                            Settings indexSettings, MultiFields multiFields, Explicit<Boolean> ignoreMalformed,
+                            CopyTo copyTo) {
+        super(simpleName, fieldType, defaultFieldType, indexSettings, multiFields, copyTo);
+        this.ignoreMalformed = ignoreMalformed;
+    }
+
+    @Override
+    protected void doMerge(Mapper mergeWith) {
+        super.doMerge(mergeWith);
+        PointFieldMapper gpfmMergeWith = (PointFieldMapper) mergeWith;
+        if (gpfmMergeWith.ignoreMalformed.explicit()) {
+            this.ignoreMalformed = gpfmMergeWith.ignoreMalformed;
+        }
+    }
+
+    @Override
+    protected String contentType() {
+        return CONTENT_TYPE;
+    }
+
+    @Override
+    protected void parseCreateField(ParseContext context, List<IndexableField> fields) throws IOException {
+        throw new UnsupportedOperationException("Parsing is implemented in parse(), this method should NEVER be called");
+    }
+
+    public static class PointFieldType extends AbstractSearchableGeometryFieldType {
+        public PointFieldType() {
+        }
+
+        PointFieldType(PointFieldType ref) {
+            super(ref);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public MappedFieldType clone() {
+            return new PointFieldType(this);
+        }
+
+        @Override
+        public ValuesSourceType getValuesSourceType() {
+            return CoreValuesSourceType.GEOPOINT;
```

Comment:
This feels trappy. I think we need to add a new `ValuesSourceType.POINT` so users can't run `geo_` aggregations on cartesian types? /cc @not-napoleon  Probably best to remove doc value support from this PR and add it as a separate feature PR.

Reply:
That should not be there. I have removed the method so it fall back to the default implementation that throws a meaningful error.